libsigner/src/libsigner.rs

```diff
@@ -55,3 +62,17 @@ pub use crate::events::{
 pub use crate::runloop::{RunningSigner, Signer, SignerRunLoop};
 pub use crate::session::{SignerSession, StackerDBSession};
 pub use crate::signer_set::{Error as ParseSignerEntriesError, SignerEntries};
+
+/// A trait for message slots used for signer communication
+pub trait MessageSlotID: Sized + Eq + Hash + Debug + Copy {
+    /// The contract identifier for the message slot in stacker db
+    fn stacker_db_contract(&self, mainnet: bool, reward_cycle: u64) -> QualifiedContractIdentifier;
+    /// All possible Message Slot values
+    fn all() -> Vec<Self>;
```

Comment:
This seems a little wasteful. Can this instead return a `&'static [Self]`, and have the implementation return a reference to a `pub const`- or a `lazy_static! { .. }`-declared slice containing all of the possible values?